### ISSUE TITLE
deploy genAiPlanner with changes won't update UI

### ISSUE THREAD

**Issue by martinadolfi** [forcedotcom/cli#3068]
### Summary
* Having a new GenAiPlanner in the code and deploying it, won't be reflected in the org UI.
* Making changes in the description of a GenAiPlanner object and deploying the change won't be reflected in the org UI.
* Strange enough, with an existing object, changing GenAiPluggin objects within the planner works.

This all has been tested with type=`Digital Channel` 

### Expected result
When org is empty and doesn't have the KDA agent created and executing the following command, I expect the planner to be created.
```bash
sf project deploy start --metadata GenAiPlanner:kda_v1
```

### Actual result
Though the CLI succeeds, the planner won't show up in the org UI.

Additionally, if the object is created by hand in the org, and pulled, changes in the descriptions have the same outcome, they can be made and pushed with the CLI but though the CLI succeeds, the change won't be shown in the UI.

### Additional information

Example repo and file https://git.soma.salesforce.com/madolfi/knowledge-driven-actions-poc/blob/master/force-app/main/default/genAiPlanners/kda_v1.genAiPlanner-meta.xml

### System Information

```
CLI:
@salesforce/cli/2.63.7 darwin-x64 node-v20.17.0

Plugin Version:
@oclif/plugin-autocomplete 3.2.6 (core)
@oclif/plugin-commands 4.1.3 (core)
@oclif/plugin-help 6.2.15 (core)
@oclif/plugin-not-found 3.2.22 (core)
@oclif/plugin-plugins 5.4.15 (core)
@oclif/plugin-search 1.2.12 (core)
@oclif/plugin-update 4.6.4 (core)
@oclif/plugin-version 2.2.15 (core)
@oclif/plugin-warn-if-update-available 3.1.19 (core)
@oclif/plugin-which 3.2.16 (core)
@salesforce/cli 2.63.7 (core)
apex 3.5.4 (core)
api 1.3.1 (core)
auth 3.6.68 (core)
data 3.7.1 (core)
deploy-retrieve 3.14.0 (core)
devops-center 1.2.24 (user)
functions 1.23.0 (user)
info 3.4.12 (core)
limits 3.3.33 (core)
marketplace 1.2.26 (core)
org 4.7.0 (core)
packaging 2.8.12 (core)
schema 3.3.36 (core)
settings 2.3.24 (core)
signups 2.5.23 (user)
sobject 1.4.42 (core)
source 3.5.21 (core)
telemetry 3.6.16 (core)
templates 56.3.24 (core)
trust 3.7.34 (core)
user 3.5.33 (core)
```

```
SF ENV. VARS.
SF_BINPATH,/Users/madolfi/.local/share/sf/client/bin/sf
SF_AUTOUPDATE_DISABLE,true
SF_DISABLE_AUTOUPDATE,true
SF_UPDATE_INSTRUCTIONS,Use "npm update --global @salesforce/cli" to update npm-based installations.
```

```
Windows: false
Shell: zsh
Channel: stable
```
---
### Diagnostics
:white_check_mark: pass - salesforcedx plugin isn’t installed
:white_check_mark: pass - you don't have any linked plugins
:white_check_mark: pass - [@salesforce/plugin-trust] can ping: https://registry.npmjs.org
:white_check_mark: pass - [@salesforce/plugin-trust] can ping: https://registry.yarnpkg.com
:white_check_mark: pass - [@salesforce/plugin-trust] can ping: https://registry.npmjs.org/
:white_check_mark: pass - using latest or latest-rc CLI version
:white_check_mark: pass - [@salesforce/plugin-deploy-retrieve] sourceApiVersion matches apiVersion
:x: warn - [@salesforce/plugin-deploy-retrieve] sourceApiVersion matches default target org max apiVersion
:white_check_mark: pass - can access: https://test.salesforce.com
:white_check_mark: pass - can access: https://appexchange.salesforce.com/services/data
:white_check_mark: pass - can access: https://developer.salesforce.com/media/salesforce-cli/sf/channels/stable/sf-win32-x64-buildmanifest
:x: fail - [@salesforce/plugin-auth] CLI supports v2 crypto

**Comment by shetzel** [comment]
@martinadolfi - thanks for reporting.  The CLI is sending the correct request to the metadata API.  This is an issue for the metadata type owner of GenAiPlanner.

**Comment by shetzel** [comment]
This is now fixed in SFDC release 252.13
